src/analytics/dsl.tsx

```diff
@@ -0,0 +1,12 @@
+export const dsl = {
+  "Follow Artist": {
+    intent: (object) => ({ artist_id: object.artist.id }),
+    success: (object) => ({ artist_id: object.artist.id }),
+    fail: (object) => ({ artist_id: object.artist.id }),
+  },
+  "Unfollow Artist": {
+    intent: (object) => ({ artist_id: object.artist.id }),
+    success: (object) => ({ artist_id: object.artist.id }),
+    fail: (object) => ({ artist_id: object.artist.id }),
+  },
+}
```

Comment:
In retrospect I wonder if this large object that mimics ARAnalytics is the best idea, because that large object sometimes becomes a bit of a hassle to maintain and is really only done because otherwise objc makes DSLs cumbersome. Additionally we probably want to add multiple analytics files at some point, so being able to call those functions from anywhere in the project is more flexible.

```js
import * as analytics from "./dsl"

analytics.on("Follow Artist", {
  intent: (object) => ({ artist_id: object.artist.id }),
  success: (object) => ({ artist_id: object.artist.id }),
  fail: (object, error) => ({ artist_id: object.artist.id }),
})
```